rktl_sim/src/simulator/sim.py

```diff
@@ -235,16 +353,90 @@ def get_ball_pose(self, add_noise=False):
 
         return pos, p.getQuaternionFromEuler([0, 0, 0])
 
+    """get the current ball velocity"""
+
     def get_ball_velocity(self):
         if self._ball_id is None:
             return None
         return p.getBaseVelocity(self._ball_id)
 
-    def reset(self):
+    """
+    Set the winner, scored, touch last variables to none
+    for the ball:
+    - Reset the ball_pos via init_ball_pos
+    - If the init_ball_pos has no default provided, randomize the ball spawn location
+        - using the paramters from above:
+        - Reset the base position and orientation of the ball in pybullet: give the ball id, ball pos, angle for the specific ball
+    
+    - Reset the ball velocity: randomize it by default
+        - reset it in pybullet via resetBaseVelocity
+        
+    For each car: 
+    - try to get the ini_pos and init_orientation if it was provided
+    - If the orientation and position were not initialized, then randomize them, and reset the car
+    """
+
+    def reset(self, spawn_bounds, car_properties):
         self.scored = False
         self.winner = None
         self.touched_last = None
+        self.spawn_bounds = spawn_bounds
+        self.reset_ball()
```

Comment:
I will leave the functionality of reseting the car properties if we would want to do that in the future. The reset of car properties will still occur however, because the car properties themselves wont change, nothing will change here. I think that its a good feature to leave in if we would want to play around with handling the rest of car properties on the simulator side.